pkg/skaffold/deploy/kubectl.go

```diff
@@ -252,6 +219,75 @@ func (k *KubectlDeployer) readRemoteManifest(ctx context.Context, name string) (
 	return manifest.Bytes(), nil
 }
 
-func (k *KubectlDeployer) Render(context.Context, io.Writer, []build.Artifact, string) error {
-	return errors.New("not yet implemented")
+func (k *KubectlDeployer) Render(ctx context.Context, out io.Writer, builds []build.Artifact, filepath string) error {
+	manifests, err := k.renderManifests(ctx, out, builds)
+
+	if err != nil {
+		return err
+	}
+
+	manifestOut := out
+	if filepath != "" {
+		f, err := os.Open(filepath)
+		if err != nil {
+			return errors.Wrap(err, "opening file for writing manifests")
+		}
+		manifestOut = bufio.NewWriter(f)
+	}
+
+	for i := 0; i < len(manifests); i++ {
+		if _, err := fmt.Fprintf(manifestOut, string(manifests[i])); err != nil {
+			return errors.Wrap(err, "writing manifests")
+		}
+		if i < len(manifests)-1 {
+			fmt.Fprintf(manifestOut, "---\n")
+		}
+	}
```

Comment:
We could use `ManifestList.String()` here instead
```suggestion
 if _, err := fmt.Fprintf(manifestOut, manifests.String()); err !-= nil{
    return errors.Wrap(err, "writing manifests")
 }
```

Reply:
good call, done